src/App.tsx

```diff
@@ -17,18 +51,42 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                status={status}
+                setStatus={setStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                !isInitialized
+                  ? <Loader />
+                  : (
+                    <>
+                      <TodoList
+                        todos={visibleTodos}
+                        selectedTodoId={todoId}
+                        selectTodo={(id: number) => {
+                          setTodoId(id);
+                        }}
+                      />
+                    </>
+                  )
+              }
             </div>
           </div>
         </div>
+        {todoId && (
+          <TodoModal
+            todo={todos.find(selectedTodo => selectedTodo.id === todoId)}
```

Comment:
Move this logic to variable

Reply:
To avoid this finding here, let's store `selectedTodo` in the state not `todoId`